src/llmcompressor/modifiers/awq/base.py

```diff
@@ -237,7 +241,11 @@ def on_start(self, state: State, event: Event, **kwargs):
         # This choice does not seem to have a meaningful impact on accuracy
         state.model.apply(disable_quantization)
 
-        self._setup_activation_cache_hooks()
+        self._setup_activation_cache_hooks(
+            offload_device=(
+                "cpu" if self.offload_cache else get_execution_device(state.model)
```

Comment:
@kylesayrs this seems to work but please confirm this is what you'd expect it to be

Reply:
This won't work for models across separate devices. For example, if the model device is gpu0, but you're capturing from a parent on gpu1. When you try to rerun samples, the devices won't match

Reply:
This is why I suggested adding an "offload_device=None" option to the IntermedaitesCache

Reply:
updated to None, with new functionality in intermediates cache to not offload if offload_device is None